src/sampler.jl

```diff
@@ -20,7 +20,7 @@ end
 function resize(h::Hamiltonian, θ::AbstractVecOrMat{T}) where {T<:AbstractFloat}
     metric = h.metric
     if size(metric) != size(θ)
-        metric = typeof(metric)(size(θ))
+        metric = getname(typeof(metric))(size(θ))
```

Comment:
this looks good overall. For clarity, maybe consider merging `getname(typeof(metric))` into `getname(metric)`, i.e. calling the `typeof` function inside `getname`?